TestStack.Dossier.Tests/BuilderBuildTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Shouldly;
+using TestStack.Dossier.Tests.Builders;
+using TestStack.Dossier.Tests.Entities;
+using Xunit;
+
+namespace TestStack.Dossier.Tests
+{
+    public class BuilderBuildTests
+    {
+        [Fact]
+        public void GivenBasicBuilder_WhenCallingBuildExplicitly_ThenReturnAnObject()
+        {
+            var builder = Builder<Customer>.CreateNew();
+
+            var customer = builder.Build();
+
+            customer.ShouldBeOfType<Customer>();
+        }
+
+        [Fact]
+        public void GivenBuilder_WhenCallingSetExplicitly_ShouldOverrideValues()
+        {
+            var builder = Builder<Customer>.CreateNew()
```

Comment:
Personally, I would find it frustrating to have to implement BuildObject for every builder that implements TestDataBuilder. Of course, I can create a base builder in my test infrastructure that implements it, so it's not a show stopper. But as someone coming from NBuilder, which didn't require that, I would find it unintuitive about why this library requires that I need to do that. Surely the whole point of a virtual method is that you have the best of both worlds - you get the default behaviour or choose to override if you want something else. I'm concerned that making it abstract, and forcing people to override it, might alienate users as you are removing the choice?